TheraLang.Web/Controllers/PageController.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TheraLang.BLL.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using TheraLang.Web.ViewModels;
+using TheraLang.Web.Extensions;
+using AutoMapper;
+using TheraLang.BLL.DataTransferObjects;
+using TheraLang.BLL.CustomTypes;
+
+namespace TheraLang.Web.Controllers
+{
+    [Route("api/pages")]
+    [ApiController]
+    public class PageController : ControllerBase
+    {
+        private readonly IPageService _pageService;
+        private readonly IUserManagementService _userManagementService;
+
+        public PageController(IPageService pageService, IUserManagementService userManagementService)
+        {
+            _pageService = pageService;
+            _userManagementService = userManagementService;
+        }
+
+        [HttpPost]
+        [Authorize]
+        public async Task<IActionResult> CreatePage([FromBody] PageViewModel pageModel)
+        {
+            var UserId = User.Claims.GetUserId();
+            if (UserId == null) return BadRequest();
+
+            if (pageModel == null) return BadRequest();
+
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<PageViewModel, PageDto>()
+            .ForMember(c => c.Content, opt => opt.MapFrom(n => new HtmlContent(n.Content))))
+                .CreateMapper();
+            var pageDto = mapper.Map<PageViewModel, PageDto>(pageModel);
+
+            await _pageService.Add(pageDto);
+            return Ok(pageDto);
+        }
+
+        [HttpPut("{id}")]
+        [Authorize]
+        public async Task<IActionResult> EditPage(int id, [FromBody] PageViewModel pageModel)
+        {
+            var UserId = User.Claims.GetUserId();
+            if (UserId == null) return BadRequest();
```

Comment:
seems like copy paste, can we do this check in one place?

Reply:
Its a question if need it at all, 'cause user needs to be authorized to access these methods what means claims contain value for UserId. So i don't think there would be case where it would be null